### PR TITLE
Update default SKU ID naming format

### LINKED ISSUE #336: Image versioning / SKU naming
Currently, the Azure Marketplace images published for capz follow this naming:

```
Offer        Publisher      Sku                     Urn                                                          Version
-----------  -------------  ----------------------  -----------------------------------------------------------  ----------
capi         cncf-upstream  k8s-1dot16-ubuntu-1804  cncf-upstream:capi:k8s-1dot16-ubuntu-1804:2019.11.04         2019.11.04
```

Then, the code fetches the `latest` version for the SKU matching the desired k8s version: 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/328
https://github.com/Azure/cluster-api-provider-azure/blob/master/cloud/defaults.go#L152

The issue with this is that there is no way to distinguish between patch versions of Kubernetes. So if a user wants to create a cluster with e.g. `1.16.1` but the latest image in the 1.16 SKU is `1.16.2`, then the default image picked will be `1.16.2` and the cluster will come up with `1.16.2`.

We have the following constraints: 
- SKU ID must consist only of lowercase, alphanumeric characters, dashes or underscores. Cannot be modified once published. Max 50 characters.
- The number of versions per SKU is limited.
- version has to be in <number>.<number>.<number> format and has to be unique per SKU

We should consider the following info for image naming:
K8s version, OS type/distro, OS SKU, CNI (CNI version?), date published

Right now we only have a k8s `1.16` image published. Once we agree on the image naming format I will proceed and publish all the missing images.

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Change the SKU ID default for upcoming published images.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #336 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Update default image SKU ID format
```
/hold until images are tested

### CODE REVIEW

File: cloud/defaults.go [hunk 1]
```diff
@@ -133,19 +133,9 @@ func GenerateOSDiskName(machineName string) string {
...
-	if strings.HasPrefix(k8sVersion, "v1.14") {
-		return "k8s-1dot14-ubuntu-1804"
-	}
-	if strings.HasPrefix(k8sVersion, "v1.13") {
-		return "k8s-1dot13-ubuntu-1804"
-	}
-	return ""
+	k8sVersion = strings.TrimPrefix(k8sVersion, "v")
+	parts := strings.SplitN(k8sVersion, ".", 3)
+	return fmt.Sprintf("k8s-%sdot%sdot%s-ubuntu-18.04", parts[0], parts[1], parts[2])
```
Comment (CecileRobertMichon) [inline cloud/defaults.go]: // todo : add unit test

File: cloud/defaults.go [hunk 2]
```diff
@@ -133,19 +133,12 @@ func GenerateOSDiskName(machineName string) string {
 
 // GetDefaultImageSKUID gets the SKU ID of the image to use for the provided version of Kubernetes.
 func getDefaultImageSKUID(k8sVersion string) string {
-	if strings.HasPrefix(k8sVersion, "v1.16") {
-		return "k8s-1dot16-ubuntu-1804"
+	k8sVersion = strings.TrimPrefix(k8sVersion, "v")
+	parts := strings.SplitN(k8sVersion, ".", 3)
+	if len(parts) != 3 {
+		return ""
```
Comment (devigned) [inline cloud/defaults.go]: why not just return an error?

File: cloud/defaults.go [hunk 3]
```diff
@@ -133,19 +133,12 @@ func GenerateOSDiskName(machineName string) string {
 
 // GetDefaultImageSKUID gets the SKU ID of the image to use for the provided version of Kubernetes.
 func getDefaultImageSKUID(k8sVersion string) string {
-	if strings.HasPrefix(k8sVersion, "v1.16") {
-		return "k8s-1dot16-ubuntu-1804"
+	k8sVersion = strings.TrimPrefix(k8sVersion, "v")
```
Comment (vincepri) [inline cloud/defaults.go]: This is what I did in GCP to parse k8s versions: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/6b1aef7ac3b53c8affad389081e8551184d9273e/cloud/services/compute/instances.go#L199-L203

File: cloud/defaults.go (hunk 2 above)
Comment (CecileRobertMichon) [reply cloud/defaults.go]: changed to return an error, I was on the edge about doing that but figured it was easier to keep the returns as-is, thanks for the nudge.

File: cloud/defaults.go (hunk 3 above)
Comment (CecileRobertMichon) [reply cloud/defaults.go]: thanks for the tip! changed to use the same thing

**Pull request merged**